core/analysis/text_token_stream.cpp

```diff
@@ -359,8 +353,8 @@ analysis::analyzer::ptr construct(icu::Locale&& locale) {
       return nullptr;
     }
 
-    return construct(options.locale.getName(), std::move(options),
-                     std::move(stopwords));
+    return construct(hashed_string_view{options.locale.getName()},
+                     std::move(options), std::move(stopwords));
```

Comment:
Is this safe? We are creating a hased_string_view on some locale name from `options`, but the next action is to `std::move` the `options` object somewhere else.

Reply:
Yes, it's safe because construct accept by rvalue reference, so there hasn't move ctor